### ISSUE TITLE
Ensure Travis fails fast

### ISSUE THREAD

**Issue by 0xazure** [0xazure/supernova#46]
In reviewing #34, I noticed [one of the build jobs](https://travis-ci.org/0xazure/supernova/jobs/470150369) failed the `clippy` step but continued on to the test and build steps anyways. This seems like needless work, and it might be more desirable to tighten up the builds feedback on PRs so the build fails immediately if one of the checks like `clippy` or `rustfmt` fails.

@SeanPrashad does Travis not fail-fast, or do we need to configure something else? We already have `fast_finish` enabled in our config:

https://github.com/0xazure/supernova/blob/66f5bc50aa6bfe8c359e684279e3c889039b8d0e/.travis.yml#L35

**Comment by biskit1** [comment]
is it related to the fact that we allow failures for `rust: nightly` so all the tests against the nightly channel will still resume?

**Comment by 0xazure** [comment]
No, the `allow_failures` for nightly means that the `nightly` job in the build matrix can fail but the entire build won't fail and e.g. merging a pull request won't be blocked. See: [Rows that are Allowed to Fail](https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail).

Also, the specific job I linked to there is for the `stable` release channel, so it isn't affected by allowing failures for `nightly`.
